### ISSUE TITLE
[Accessibility]:  `Statement of truth` heading needs to be flexible in the component

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/vets-design-system-documentation#5945]
### Point of contact

jeana-adhoc

### How severe do you think this issue is?

High

### Severity rationale

Users are not blocked from completing their task, but it does provide a confusing experience for users.

This is a violation of WCAG 1.3.1

### VA 508 Office

N/A

### URL where issue occurs

https://design.va.gov/storybook/?path=/story/uswds-va-statement-of-truth--default

### Component or pattern (if applicable)

va-statement-of-truth

### Environment details

All - this is not dependent upon screen readers or browsers

### Steps to reproduce

1. Load storybook
2. Go to https://design.va.gov/storybook/?path=/story/uswds-va-statement-of-truth--default

### Observed behavior

See there is no way to change the heading level.

### Expected behavior / Acceptance criteria

Consumers can adjust the heading level in the component

### Supporting evidence (screenshots, recordings, code snippets)

In minimal form flow when the statement of truth is used, the heading needs to be changed to an `<h2>`

<img width="1021" height="310" alt="Image" src="https://github.com/user-attachments/assets/4f5b3ad2-95a3-4363-91d7-c0403b2791a3" />

### Type of issue

- [ ] Axe-core
- [ ] Cognitive | user flows & navigating
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Content style guide
- [ ] Focus
- [x] Headings
- [ ] Keyboard
- [ ] Link and button labels
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [ ] Screenreader | assistive tech & device support
- [ ] Semantic HTML
- [ ] URL standards
- [ ] Zoom or Magnification
- [ ] Something else
